src/commands/deployment/correlate.ts

```diff
@@ -0,0 +1,182 @@
+import chalk from 'chalk'
+import {Command, Option} from 'clipanion'
+import simpleGit from 'simple-git'
+
+import {getCISpanTags} from '../../helpers/ci'
+import {gitRepositoryURL, gitLocalCommitShas, gitCurrentBranch} from '../../helpers/git/get-git-data'
+import {Logger, LogLevel} from '../../helpers/logger'
+import {retryRequest} from '../../helpers/retry'
+import {CI_PROVIDER_NAME, CI_ENV_VARS, GIT_REPOSITORY_URL, GIT_SHA} from '../../helpers/tags'
+import {getApiHostForSite, getRequestBuilder} from '../../helpers/utils'
+
+/**
+ * This command is a wrapper around the datadog-ci tag command, allowing customers to mark CI jobs
+ * as deployments and setting specific properties, like the environment or the revision in a simple way.
+ */
+export class DeploymentCorrelateCommand extends Command {
+  public static paths = [['deployment', 'correlate']]
+
+  public static usage = Command.Usage({
+    category: 'CI Visibility',
+    description: 'Correlate gitOps CD deployments with CI pipelines.',
+    details: `
+      This command will correlate the pipeline with a gitOps CD deployment.\n
+      It does not work for every setup, see README for details.
+    `,
+    examples: [['Correlate an argoCD deployment', 'datadog-ci deployment correlate --provider argocd']],
+  })
+
+  private cdProviderParam = Option.String('--provider')
+  private cdProvider!: string
+  private supportedCDProviders: string[] = ['argocd'] // update the README if changing this
+  private configurationRepo = Option.String('--config-repo')
+  private dryRun = Option.Boolean('--dry-run', false)
+
+  private config = {
+    apiKey: process.env.DATADOG_API_KEY || process.env.DD_API_KEY,
+  }
+
+  private logger: Logger = new Logger((s: string) => this.context.stdout.write(s), LogLevel.INFO)
+
+  public async execute() {
+    if (!this.config.apiKey) {
+      this.logger.error(
+        `Neither ${chalk.red.bold('DATADOG_API_KEY')} nor ${chalk.red.bold('DD_API_KEY')} is in your environment.`
+      )
+
+      return 1
+    }
+
+    if (!this.validateCDProviderParam()) {
+      return 1
+    }
+
+    const tags = getCISpanTags() || {}
+
+    if (!this.validateTags(tags)) {
```

Comment:
This is something else on which I would rely on API validation directly 😬

Reply:
I would leave this one since it just checks whether there is a commit SHA and repo URL. We will always need this and we can prevent customers from doing bad requests before sending the request